pilot/pkg/serviceregistry/mcp/controller.go

```diff
@@ -192,6 +192,23 @@ func (c *controller) Apply(change *sink.Change) error {
 
 	c.configStoreMu.Lock()
 	prevStore = c.configStore[kind]
+	// Incremental update when received incremental change
+	if change.Incremental {
+
+		//Although it is not a deep copy, there is no problem because the config will not be modified
+		prevCache := make(map[string]map[string]*model.Config, len(prevStore))
+		for namespace, namedConfig := range prevStore {
+			prevCache[namespace] = make(map[string]*model.Config, len(namedConfig))
+			for name, config := range namedConfig {
+				prevCache[namespace][name] = config
+			}
+		}
+		prevStore = prevCache
```

Comment:
What is this block doing? `prevStore` is never used.

Reply:
When kind is ServiceEntry, prevStroe will be used in the following code

![image](https://user-images.githubusercontent.com/41314762/76375425-c331b580-6380-11ea-869f-1877742c63aa.png)

Actually this block is only needed when kind is ServiceEntry, but I'm not sure if it is suitable to hard code this judgment condition.